mmf/models/base_model.py

```diff
@@ -213,9 +213,13 @@ def from_pretrained(cls, model_name, *args, **kwargs):
         assert (
             model_cls == cls
         ), f"Incorrect pretrained model key {model_name} for class {cls.__name__}"
-        output = load_pretrained_model(model_name, *args, **kwargs)
-        config, checkpoint = output["config"], output["checkpoint"]
 
+        if "from_file" in kwargs:
+            output = load_pretrained_model(kwargs["from_file"], *args, **kwargs)
+        else:
+            output = load_pretrained_model(model_name, *args, **kwargs)
```

Comment:
The argument `model_name` should be able to take a file path, something like `model_name_or_file_path`. That will be more convenient to use and also similar to how other frameworks handle loading pretrained models. 

https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_utils.py#L743

Reply:
https://github.com/facebookresearch/mmf/blob/master/mmf/models/mmbt.py#L617

It looks like because of the above code, we'd need to specify what the model name is here rather than overriding the filepath together with the model. 

On the other hand, I think we should not have hardcoding the model in code like this, perhaps we should add the flag into the model config? @apsdehal

Reply:
I agree it seems a bit hacky right now with hardcoded model name. I think in case of MMBT HM interface, we can pass a kwarg like "model_interface=True" that will return a `MMBTGridHMInterface` object for this specific case.

Reply:
I think that use case is fine to ignore. If user is passing the file path directly one can return back the model in else condition. Please fix that in PR directly. Otherwise, Vedanuj said is important and should be implemented that way. Also, this usually is folder path not a file path where the checkpoint and config is stored. Can you fix `load_pretrained_model` so that it can directly take in a checkpoint and expect the checkpoint to have "config" attribute? Check `load_pretrained_model`'s implementation for more details.

Reply:
Hi @apsdehal and @vedanuj, I updated the PR, let me know if you have more questions. Thanks for the feedback!